### PR TITLE
Benchmark script improvements

### PR DESCRIPTION
**Pull request by domanchi**
### Summary

Building off https://github.com/Yelp/detect-secrets/pull/181, this adds the following features:

- machine consumable output, by default. (`--pretty` preserves the human readable table)
- you can select which plugins you want to test
- timeout functionality, so scripts won't run forever
- customizing number of iterations, so you can get more accurate results

### Example Output

```
$ python scripts/benchmark.py --plugin PrivateKeyDetector --plugin Base64HighEntropyString -n 5 | jq
Running performance tests on: PrivateKeyDetector, Base64HighEntropyString
for: ['~/pg/detect-secrets']
{
    "Base64HighEntropyString": 0.8232977999992727,
    "PrivateKeyDetector": 0.6896803999989061
}
```

```
$ python scripts/benchmark.py --plugin Base64HighEntropyString --harakiri 0.5 --pretty
Running performance tests on: Base64HighEntropyString
for: ['~/pg/detect-secrets']
------------------------------------------
plugin                              time
------------------------------------------
Base64HighEntropyString   Timeout exceeded!
------------------------------------------
```

### CODE REVIEW

File: scripts/benchmark.py [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+        default=None,       # needs to be None, otherwise append won't work as expected
+        choices=plugins,
+        action='append',
+        help=(
+            'Specifies a plugin to test. May provide multiple values. '
+            'Defaults to all.'
+        ),
+    )
+    parser.add_argument(
+        '--harakiri',
```
Comment (KevinHock) [inline scripts/benchmark.py]: lol

File: scripts/benchmark.py [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+        help=(
+            'Specifies an upper bound for the number of seconds to wait '
+            'per execution.'
+        ),
+    )
+    parser.add_argument(
+        '-n',
+        '--num-iterations',
+        default=1,
+        # TODO: assert non-negative
```
Comment (KevinHock) [inline scripts/benchmark.py]: [This seems like the way to do that](https://stackoverflow.com/questions/14117415/in-python-using-argparse-allow-only-positive-integers)

File: scripts/benchmark.py [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+            'Specifies the number of times to run the test. '
+            'Results will be averaged over this value.'
+        ),
+    )
+
+    args = parser.parse_args()
+    if not args.filenames:
+        args.filenames = [
+            os.path.realpath(
+                os.path.join(
```
Comment (KevinHock) [inline scripts/benchmark.py]: Super nit: variable names could maybe help readability here

File: scripts/benchmark.py (hunk 2 above)
Comment (domanchi) [reply scripts/benchmark.py]: Whoops. Forgot to remove this lol.

Yeah, we're not breaking new ground here. We've done this before: https://github.com/Yelp/detect-secrets/blob/master/detect_secrets/core/usage.py#L396

File: scripts/benchmark.py (hunk 1 above)
Comment (domanchi) [reply scripts/benchmark.py]: FWIW, I copied uwsgi's syntax for this: https://uwsgi-docs.readthedocs.io/en/latest/Glossary.html

**Pull request merged**